xlators/mgmt/glusterd/src/glusterd-op-sm.c

```diff
@@ -2973,12 +3105,23 @@ glusterd_op_set_volume(dict_t *dict, char **errstr)
         if (global_opt) {
             cds_list_for_each_entry(voliter, &priv->volumes, vol_list)
             {
+                /* Check dependency of options before setting them */
+                ret = glusterd_dependency_chain_check(voliter, key, value,
+                                                      errstr);
+                if (ret)
+                    goto out;
+
```

Comment:
The distinguishing between the global option and the per-vol option is done above in a well-organized way, before reaching here. So, doing the same thing again will add an overhead I think. So,  instead of doing the check again I called the function here to check the dependency for the global opt as well.